File: daily_python/2894_Divisible_and_Non_divisible_Sums_Difference.py
import os
import sys

class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        group = n // m
        return (1 + n) * n // 2 - m * (1 + group) * group

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            m = int(lines[0].split('=')[1].strip()[:-1])
            n = int(lines[1].split('=')[1].strip()[:-1])
            dataset.append((m, n))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for m, n in dataset:
        results.append(solution.differenceOfSums(m, n))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2025/2894_Divisible_and_Non_divisible_Sums_Difference.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int differenceOfSums(int n, int m) {
        // [1 2 3 ... (m-1) -m] + ..... + [(n/m-1)*m+1 .... -(n / m * m)] + [.... n]
        // [1 2 .... n] - 2 * [m 2*m .... n/m*m]
        // (1 + ... + n) - 2 * m(1 + ... + group)
        int group = n / m;
        return (1 + n) * n / 2 - (1 + group) * group * m;
    }
};

int main(){
    int n = 10;
    int m = 3;
    // int n = 5;
    // int m = 6;
    // int n = 5;
    // int m = 1;
    Solution S;
    cout << S.differenceOfSums(n, m) << endl;
    return 0;
}